### ISSUE TITLE
🐛 [firebase_messaging] plugins not working on onBackgroundMessage

### ISSUE THREAD

**Issue by imSaharukh** [firebase/flutterfire#8872]
## Bug report

flutter plugins are not working on onBackgroundMessage

### Sample project

[GIT REPO](https://github.com/imSaharukh/pluings-on-backround-notification)

## Additional context

I'm using assets_audio_player to play a sound when a fcm notification received 

onBackgroundMessage handler is

```
Future myBackgroundMessageHandler(RemoteMessage message) async {
  await Firebase.initializeApp();
  final assetsAudioPlayer = AssetsAudioPlayer();

  assetsAudioPlayer.open(
    Audio.network(
        "https://www.soundhelix.com/examples/mp3/SoundHelix-Song-2.mp3"),
  );

  print("sound played");
  return true;
}

```

but not working when the app is the background 
it gives the error given below

```
I/flutter (13448): MissingPluginException(No implementation found for method stop on channel assets_audio_player)
I/flutter (13448): MissingPluginException(No implementation found for method open on channel assets_audio_player)
E/flutter (13448): [ERROR:flutter/lib/ui/ui_dart_state.cc(198)] Unhandled Exception: MissingPluginException(No implementation found for method open on channel assets_audio_player)
E/flutter (13448):
```

### Flutter doctor

<details><summary>Click To Expand</summary>

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.0.0, on macOS 12.4 21F79 darwin-arm, locale
    en-BD)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 13.3)
[✓] Chrome - develop for the web
[✓] Android Studio
[✓] VS Code (version 1.68.0)
[✓] Connected device (3 available)
    ! Error: S A’s iPhone is not connected. Xcode will continue when S A’s
      iPhone is connected. (code -13)
[✓] HTTP Host Availability

• No issues found!
```

</details>

### Flutter dependencies

<details><summary>Click To Expand</summary>

```
Dart SDK 2.17.0
Flutter SDK 3.0.0
bg_dart_code_fcm 1.0.0+1

dependencies:
- assets_audio_player 3.0.4+3 [flutter rxdart uuid http path_provider assets_audio_player_web]
- cupertino_icons 1.0.5
- firebase_core 1.17.1 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_messaging 11.4.1 [firebase_core firebase_core_platform_interface firebase_messaging_platform_interface firebase_messaging_web flutter meta]
- flutter 0.0.0 [characters collection material_color_utilities meta vector_math sky_engine]
- flutter_local_notifications 9.6.0 [clock flutter flutter_local_notifications_linux flutter_local_notifications_platform_interface timezone]

dev dependencies:
- flutter_lints 2.0.1 [lints]
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters charcode collection matcher material_color_utilities meta source_span stream_channel string_scanner term_glyph]

transitive dependencies:
- args 2.3.1
- assets_audio_player_web 3.0.4+3 [flutter flutter_web_plugins]
- async 2.8.2 [collection meta]
- boolean_selector 2.1.0 [source_span string_scanner]
- characters 1.2.0
- charcode 1.3.1
- clock 1.1.0
- collection 1.16.0
- crypto 3.0.2 [typed_data]
- dbus 0.7.5 [args ffi meta xml]
- fake_async 1.3.0 [clock collection]
- ffi 2.0.0
- file 6.1.2 [meta path]
- firebase_core_platform_interface 4.4.0 [collection flutter meta plugin_platform_interface]
- firebase_core_web 1.6.4 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- firebase_messaging_platform_interface 3.5.1 [firebase_core flutter meta plugin_platform_interface]
- firebase_messaging_web 2.4.1 [firebase_core firebase_core_web firebase_messaging_platform_interface flutter flutter_web_plugins js meta]
- flutter_local_notifications_linux 0.5.0+1 [flutter flutter_local_notifications_platform_interface dbus path xdg_directories]
- flutter_local_notifications_platform_interface 5.0.0 [flutter plugin_platform_interface]
- flutter_web_plugins 0.0.0 [flutter js characters collection material_color_utilities meta vector_math]
- http 0.13.4 [async http_parser meta path]
- http_parser 4.0.1 [collection source_span string_scanner typed_data]
- js 0.6.4
- lints 2.0.0
- matcher 0.12.11 [stack_trace]
- material_color_utilities 0.1.4
- meta 1.7.0
- path 1.8.1
- path_provider 2.0.11 [flutter path_provider_android path_provider_ios path_provider_linux path_provider_macos path_provider_platform_interface path_provider_windows]
- path_provider_android 2.0.14 [flutter path_provider_platform_interface]
- path_provider_ios 2.0.9 [flutter path_provider_platform_interface]
- path_provider_linux 2.1.7 [ffi flutter path path_provider_platform_interface xdg_directories]
- path_provider_macos 2.0.6 [flutter path_provider_platform_interface]
- path_provider_platform_interface 2.0.4 [flutter platform plugin_platform_interface]
- path_provider_windows 2.1.0 [ffi flutter path path_provider_platform_interface win32]
- petitparser 5.0.0 [meta]
- platform 3.1.0
- plugin_platform_interface 2.1.2 [meta]
- process 4.2.4 [file path platform]
- rxdart 0.27.4
- sky_engine 0.0.99
- source_span 1.8.2 [collection path term_glyph]
- stack_trace 1.10.0 [path]
- stream_channel 2.1.0 [async]
- string_scanner 1.1.0 [charcode source_span]
- term_glyph 1.2.0
- test_api 0.4.9 [async boolean_selector collection meta source_span stack_trace stream_channel string_scanner term_glyph matcher]
- timezone 0.8.0 [path]
- typed_data 1.3.1 [collection]
- uuid 3.0.6 [crypto]
- vector_math 2.1.2
- win32 2.7.0 [ffi]
- xdg_directories 0.2.0+1 [meta path process]
- xml 6.1.0 [collection meta petitparser]
```
</details>

**Comment by maheshmnj** [comment]
Hi @imSaharukh, The error seems to point to an issue in a third party plugin `assets_audio_player`, I would suggest filing the issue in the dedicated repo https://github.com/florent37/Flutter-AssetsAudioPlayer

>[ERROR:flutter/lib/ui/ui_dart_state.cc(198)] Unhandled Exception: MissingPluginException(No implementation found for method open on channel assets_audio_player)

**Comment by realh** [comment]
I don't think it's the fault of assets_audio_player, because I get the same error trying to call platform-specific code directly from my onBackgroundMessage handler. I set up the MethodChannel in my own code, not in a plugin, and the message handler doesn't refer to any plugins. The platform-specific code is currently very simple; it doesn't try to call back to any Dart. So there must be a bug in MethodChannel or FCM background handlers that's producing a misleading error message.

 I noticed that the Kotlin/Java import is this:
```
import io.flutter.plugin.common.MethodChannel
```
implying some part of the framework thinks MethodChannel is in a plugin. However, it either isn't a plugin, or it's built-in (there is a "methodchannel" plugin in pub, but it seems to just be someone's starter project). Maybe this has something to do with the error?
